torchvision/csrc/io/image/cpu/decode_jpeg.cpp

```diff
@@ -139,12 +203,28 @@ torch::Tensor decode_jpeg(const torch::Tensor& data, ImageReadMode mode) {
   auto tensor =
       torch::empty({int64_t(height), int64_t(width), channels}, torch::kU8);
   auto ptr = tensor.data_ptr<uint8_t>();
+  torch::Tensor temp_tensor;
+  if (cmyk_to_rgb) {
+    temp_tensor = torch::empty({int64_t(width), 4}, torch::kU8);
+  }
+
   while (cinfo.output_scanline < cinfo.output_height) {
     /* jpeg_read_scanlines expects an array of pointers to scanlines.
      * Here the array is only one element long, but you could ask for
      * more than one scanline at a time if that's more convenient.
      */
-    jpeg_read_scanlines(&cinfo, &ptr, 1);
+    if (cmyk_to_rgb) {
+      auto temp_buffer = temp_tensor.data_ptr<uint8_t>();
```

Comment:
Could this be called outside of the `while` loop i.e. just below line 208? I'm thinking of someting like


```Cpp
  uint8_t * temp_buffer = nullptr;
  if (cmyk_to_rgb) {
    auto temp_tensor = torch::empty({int64_t(width), 4}, torch::kU8);
    temp_buffer = temp_tensor.data_ptr<uint8_t>();
  }
```

(I'm not sure it changes anything for perf)

Reply:
I think modern compilers wont bother with this local variable representing just a pointer.